eng/versioning.targets

```diff
@@ -29,10 +29,23 @@
     </AssemblyAttribute>
   </ItemGroup>
 
+<!-- I couldn't think of anything better than such a workaround. There should be good MSBUILD magic for it. -->
```

Comment:
@MaximLipnin found a way to simplify this, we just need to introduce a target:

```xml
  <ItemGroup>
    <_unsopportedOSPlatforms Include="$(UnsupportedOSPlatforms)" />
  </ItemGroup>

  <Target Name="AddUnsoppertedOSPlatformAttribute" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild">
    <ItemGroup>
      <AssemblyAttribute Include="System.Runtime.Versioning.UnsupportedOSPlatform" Condition="'@(_unsopportedOSPlatforms)' != ''">
        <_Parameter1>%(_unsopportedOSPlatforms.Identity)</_Parameter1>
      </AssemblyAttribute>
    </ItemGroup>
  </Target>
```

Reply:
Thank you Santiago!